task/control-plane-agent/src/mgs_common.rs

```
@@ -143,14 +153,153 @@ impl MgsCommon {
+                    .request_reset();
+                // If `request_reset()` returns,
+                // something has gone very wrong.
+                panic!();
+            }
+            SpComponent::ROT => {
+                // mgs_{gimlet,psc,sidecar}.rs deal with any board specific
+                // reset strategy. Here we take care of common SP and RoT cases.
+            }
+            _ => return Err(SpError::RequestUnsupportedForComponent),
+        };
+        // We're dealing with  RoT targets at this point.
+        match update.sprot_task().reset() {
+            Err(SprotError::RspTimeout) => {
+                // This is the expected error if the reset was successful.
+                // It could be that the RoT is out-to-lunch for some other
+                // reason though.
+                // Things for upper layers to do:
+                //   - Check a boot nonce type thing to see if we are in a
+                //     new session.
+                //   - Check that the expected image is now running.
+                //     (Management plane should do that.)
+                //   - Enable staged updates where we don't automatically
+                //     reset after writing an image.
+                ringbuf_entry!(Log::RotReset {
+                    err: SprotError::RspTimeout
+                });
+                Ok(())
+            }
+            Err(err) => {
+                // Some other error occurred.
+                // Update is all-or-nothing at the moment.
+                // The control plane can try to reset the RoT again or it
+                // can start the update process all over again.  We should
+                // be able to make incremental progress if there is some
+                // bug/condition that is degrading SpRot communications.
+                ringbuf_entry!(Log::RotReset { err });
+                Err(SpError::ComponentOperationFailed(err.into()))
+            }
+            Ok(()) => Ok(()),
```

Should we log an error in this case, since we expected `Err(RspTimeout)`?

---

I logged an error here, but I'm wondering if we should actually also return an error. That would require a change in gateway-messages, so I didn't do it yet.